client/app/bundles/course/admin/pages/AnnouncementsSettings/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+
+import { AnnouncementsSettingsData } from 'types/course/admin/announcements';
+import translations from 'lib/translations/form';
+import useTranslation from 'lib/hooks/useTranslation';
+import { FormEmitter } from 'lib/components/form/Form';
+import LoadingIndicator from 'lib/components/LoadingIndicator';
+import AnnouncementsSettingsForm from './AnnouncementsSettingsForm';
+import {
+  fetchAnnouncementsSettings,
+  updateAnnouncementsSettings,
+} from './operations';
+import { useOptionsReloader } from '../../components/SettingsNavigation';
+
+const AnnouncementsSettings = (): JSX.Element => {
+  const reloadOptions = useOptionsReloader();
+  const { t } = useTranslation();
+  const [settings, setSettings] = useState<AnnouncementsSettingsData>();
+  const [form, setForm] = useState<FormEmitter>();
+  const [submitting, setSubmitting] = useState(false);
+
+  useEffect(() => {
+    fetchAnnouncementsSettings().then(setSettings);
+  }, []);
+
+  if (!settings) return <LoadingIndicator />;
+
+  const submit = (data: AnnouncementsSettingsData): void => {
```

Comment:
Lets name this handleSubmit? I think we may need to add a linter for this (could be now or next time)..
https://jaketrent.com/post/naming-event-handlers-react/

Reply:
Thanks for sharing this useful article. I have prefixed methods in all `admin/*/index.tsx` that handles posts with `handle`.